unison-cli/src/Unison/Codebase/Transcript/Runner.hs

```diff
@@ -204,12 +205,23 @@ run isTest verbosity dir codebase runtime sbRuntime nRuntime ucmVersion baseURL
           -- We shorten the terminal width, because "Transcript" manages a 2-space indent for output lines.
           Pretty.toPlain (terminalWidth - 2) line
 
-      maybeDieWithMsg :: String -> IO ()
+      maybeDieWithMsg :: Pretty.Pretty Pretty.ColorText -> IO ()
       maybeDieWithMsg msg = do
-        errOk <- readIORef allowErrors
-        if errOk
-          then writeIORef hasErrors True
-          else dieWithMsg msg
+        liftIO $ writeIORef hasErrors True
+        liftIO (liftA2 (,) (readIORef allowErrors) (readIORef expectFailure)) >>= \case
+          (False, False) -> liftIO . dieWithMsg $ Pretty.toPlain terminalWidth msg
+          (True, True) ->
+            transcriptFailure
+              out
+              ( Text.pack . Pretty.toPlain terminalWidth $
+                  "The stanza above previously had an incorrect successful result, but now fails with"
+                    <> "\n"
+                    <> Pretty.border 2 msg
+                    <> "\n"
+                    <> "if this is the expected result, remove `:bug`, otherwise remove `:error`."
```

Comment:
I think having four terms is too many, I wonder if we can do it with no more than two.
Yes to "is the error the error we're actually expecting?" that's where I was heading with the "open a new ticket if needed".

I'd be satisfied with this, wdyt:

```suggestion
                  "The stanza above marked with `:error :bug` is now failing with"
                    <> "\n"
                    <> Pretty.border 2 msg
                    <> "\n"
                    <> "so you can remove `:bug`, "close" any appropriate Github issues, and/or open a new issue if the error message is still not correct."
```

Reply:
Ok, I changed the phrasing a bit again, so see what you think in the latest commits.

I also realized that the only tests added were for actual outstanding bugs, so I added an additional transcript that will continue testing the “positive” cases after those outstanding bugs are fixed, as well as two error tests for fixed bugs (which show the messages we’ve been fiddling with).